perfkitbenchmarker/providers/azure/azure_network.py

```diff
@@ -21,264 +21,284 @@
+    create_cmd = [azure.AZURE_PATH,
                   'storage',
                   'account',
                   'create',
-                  '--affinity-group=%s' % self.affinity_group_name,
-                  '--type=%s' % self.storage_type,
-                  self.name]
+                  '--kind', self.kind,
+                  '--sku-name', self.storage_type,
+                  self.name] + self.resource_group.args
+    if self.location:
+      create_cmd.extend(
+          ['--location', self.location])
+    if self.kind == 'BlobStorage':
+      create_cmd.extend(
+          ['--access-tier', self.access_tier])
     vm_util.IssueCommand(create_cmd)
 
+  def _PostCreate(self):
+    """Get our connection string and our keys."""
+
+    stdout, _ = vm_util.IssueRetryableCommand(
+        [azure.AZURE_PATH, 'storage', 'account', 'connectionstring', 'show',
+         '--json',
+         self.name] + self.resource_group.args)
+
+    response = json.loads(stdout)
+    self.connection_string = response['string']
+    # Connection strings are always represented the same way on the
+    # command line.
+    self.connection_args = ['--connection-string', self.connection_string]
+
+    stdout, _ = vm_util.IssueRetryableCommand(
+        [azure.AZURE_PATH, 'storage', 'account', 'keys', 'list',
+         '--json',
+         self.name] + self.resource_group.args)
+
+    response = json.loads(stdout)
+    # A new storage account comes with two keys, but we only need one.
+    assert response[0]['permissions'] == 'Full'
+    self.key = response[0]['value']
+
   def _Delete(self):
     """Deletes the storage account."""
-    delete_cmd = [AZURE_PATH,
-                  'storage',
-                  'account',
-                  'delete',
-                  '--quiet',
-                  self.name]
-    vm_util.IssueCommand(delete_cmd)
+    vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'storage', 'account', 'delete',
+         '--quiet',
+         self.name] + self.resource_group.args)
 
   def _Exists(self):
     """Returns true if the storage account exists."""
-    show_cmd = [AZURE_PATH,
-                'storage',
-                'account',
-                'show',
-                '--json',
-                self.name]
-    stdout, _, _ = vm_util.IssueCommand(show_cmd, suppress_warning=True)
-    try:
-      json.loads(stdout)
-    except ValueError:
-      return False
-    return True
+    _, _, retcode = vm_util.IssueCommand(
+        [azure.AZURE_PATH, 'storage', 'account', 'show',
+         '--json',
+         self.name] + self.resource_group.args,
+        suppress_warning=True)
+
+    return retcode == 0
 
 
 class AzureVirtualNetwork(resource.BaseResource):
   """Object representing an Azure Virtual Network."""
 
-  def __init__(self, name):
+  def __init__(self, location, name):
     super(AzureVirtualNetwork, self).__init__()
     self.name = name
+    self.resource_group = GetResourceGroup()
+    self.address_space = '10.0.0.0/8'
```

Comment:
There will be one virtual network per zone, and it will include all of the resources for that zone. `10.0.0.0/8` is the largest possible address space for a virtual network, so it's sort of logical to just take it all. Would you consider that a justification? (And virtual networks in different zones have distinct address spaces, so there shouldn't be an overlap issue.)